docs/packaging-an-operator.md

```diff
@@ -0,0 +1,241 @@
+```
+
+#### Defining APIs [Required]
+
+By definition, operators are programs that can talk to the Kubernetes API. Often, they are also programs that *extend* the Kubernetes API, by providing an interface via `CustomResourceDefinition`s or, less frequently, `APIService`s. 
+
+##### Owned APIs
+
+Exactly which APIs are used and which APIs are watched or owned is important metadata for OLM. OLM uses this information to determine if dependencies are met and ensure that no two operators fight over the same resources in a cluster.
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: memcached-operator.v0.10.0
+spec:
+  # ... 
+  customresourcedefinitions:
+    owned:
+    # a list of CRDs that this operator owns 
+      # name is the metadata.name of the CRD
+    - name: cache.example.com
+      # version is the version of the CRD (one per entry)
+      version: v1alpha1
+      # spec.names.kind from the CRD
+      kind: Memcached
+```
+
+##### Required APIs
+
+Similarly, there is a section `spec.customresourcedefinitions.required`, where dependencies can be specified. The operators that provide those APIs will be discovered and installed by OLM if they have not been installed.
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: other-operator.v1.0
+spec:
+  # ... 
+  customresourcedefinitions:
+    required:
+    # a list of CRDs that this operator requires
+    - name: cache.example.com
+      version: v1alpha1
+      kind: Memcached
+```
+
+Dependency resolution and ownership is discussed more in depth in our [registry documentation]().
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: memcached-operator.v0.10.0
+spec:
+  # ... 
+  customresourcedefinitions:
+    owned:
+    # a list of CRDs that this operator owns 
+      # name is the metadata.name of the CRD
+    - name: cache.example.com
+      # version is the version of the CRD (one per entry)
+      version: v1alpha1
+      # spec.names.kind from the CRD
+      kind: Memcached
+```
+
+##### NativeAPIs [recommended]
+
+There are often cases where you wish to depend on an API that is either provided natively by the platform (i.e. `Pod`) or sometimes by another operator that is outside the control of OLM.
+
+In those cases, those dependencies can be described in the CSV as well, via `nativeAPIs`
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: other-operator.v1.0
+spec:
+  nativeAPIs:
+  - group: ""
+    version: v1
+    kind: Pod
+```
+
+The absence of any required `nativeAPIs` from a cluster will pause the installation of the operator, and `OLM` will write a status into the `CSV` indicating the missing APIs. 
+
+TODO: example status
```

Comment:
Do we want to merge these TODOs ?

Reply:
I think so, so that it's clear that these are still in progress.